### PR TITLE
Logger: Use serpent instead of dump

### PR DESCRIPTION
**Pull request by NiLuJe**
A.K.A. "Snakes, why did it have to be snakes?!".

* Persist: support serpent, and use by default over dump (as we assume consistency > readability in Persist).
* Logger/Dbg: Use serpent instead of dump to dump tables (it's slightly more compact, honors __tostring, and will tag tables with their ref, which can come in handy when debugging).
* Dbg: Don't duplicate Logger's log function, just use it directly.
* Fontlist/ConfigDialog: Use serpent for the debug dump.
* Call `os.setlocale(C, "numeric")` on startup instead of peppering it around dump calls. It's process-wide, so it didn't make much sense.
* Trapper: Use LuaJIT's serde facilities instead of dump. They're more reliable in the face of funky input, much faster, and in this case, the data never makes it to human eyes, so a human-readable format didn't gain us anything.

TL;DR: The only remaining users of dump are ReaderHistory & LuaSettings & friends. (Mostly not to rock the boat too much ;p).

Depends on https://github.com/koreader/koreader-base/pull/1527

(Will require a minor rebase after #9586).

- - -
This change is [Reviewable](https://reviewable.io/reviews/koreader/koreader/9588)

### CODE REVIEW

**Comment by zwim** [comment]
I admire Paul Kulchenko's work. 
Thanks for the PR. 
It reminded me to support ZeroBrane.

File: frontend/ui/trapper.lua [hunk 1]
```diff
@@ -674,7 +673,11 @@ function Trapper:dismissableRunInSubprocess(task, trap_widget_or_string, task_re
     end
     -- return what we got or not to our caller
     if ret_values then
-        return completed, unpack(ret_values)
+        if ret_n then
+            return completed, unpack(ret_values, 1, ret_n)
```
Comment (poire-z) [inline frontend/ui/trapper.lua:677]: unpack() has no issue with "key"=value (non idx) in the table ? and will just ignore your `ret_values.n` ? (just asking, no idea)

Comment (NiLuJe) [reply frontend/ui/trapper.lua:677]: `unpack` only operates on the array part of the table, so the n (or any other hash) are skipped. (This is why the actual return values are wrapped in an array).

Comment (NiLuJe) [reply frontend/ui/trapper.lua:677]: e.g., 

```
> t = { n = 2, [1] = 1, [2] = 2, foo = "bar" }
> print(unpack(t, 1, t.n))
1       2
> t = { foo = bar }
> print(unpack(t))

> 
```

**Review by NiLuJe — COMMENTED** [review]
> Reviewed 11 of 14 files at r1, 4 of 4 files at r2, 4 of 4 files at r3, 3 of 3 files at r4, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/koreader/koreader/9588)* status: all files reviewed, 1 unresolved discussion (waiting on @poire-z)

**Comment by poire-z** [comment]
> serpent supports serializing functions ;)

(I have a vague memory that at some point, I have written (or just thought? or moved on?) "good, we can do/keep that because dump() - or something else - won't serialize functions"  - but I can't find anything greping for 'dump(' or persist or functions... so, we'll see :)

**Comment by Frenzie** [comment]
Same.

Btw, didn't we discuss serpent a few years ago and decided against it? (Maybe just for all the work involved in switching.)

**Comment by NiLuJe** [comment]
That doesn't ring a bell (I know I *thought* about it, but didn't make the jump at the time).

FWIW, `dump` just squishes functions w/ `tostring`; and we only use it for (essentially), settings, so I'm not sure there's any potential issue?

**Comment by Frenzie** [comment]
Talking about things like logger potentially greatly increasing output; not so much settings type stuff.

> That doesn't ring a bell (I know I'd thought about it before, but didn't make the jump at the time).

Maybe it was just me thinking about it to myself, lol

**Comment by NiLuJe** [comment]
Oh, then it's entirely harmless: I'll just disable function serialization in logger ;).

**Review by NiLuJe — COMMENTED** [review]
> Reviewed 3 of 3 files at r5, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/koreader/koreader/9588)* status: all files reviewed, 1 unresolved discussion (waiting on @poire-z)

**Comment by poire-z** [comment]
Go ahead, we'll see - or we won't see anything :)
